client/src/main/java/org/folio/shared/index/util/XmlJsonUtil.java

```diff
@@ -0,0 +1,362 @@
+  }
+
+  private static String encodeXmlText(String s) {
+    StringBuilder res = new StringBuilder();
+    for (int i = 0; i < s.length(); i++) {
+      char c = s.charAt(i);
+      switch (c) {
+        case '<':
+          res.append("&lt;");
+          break;
+        case '>':
+          res.append("&gt;");
+          break;
+        case '&':
+          res.append("&amp;");
+          break;
+        case '\"':
+          res.append("&quot;");
+          break;
+        case '\'':
+          res.append("&apos;");
+          break;
+        default:
+          res.append(c);
+      }
+    }
+    return res.toString();
+  }
+
+  /**
+   * Returns XML serialized document for node in XML.
+   *
+   * <p>This method does not care about namespaces. Only elements (local names), attributes
+   * and text is dealt with.
+   *
+   * @param event event type for node that begins the subdocument
+   * @param reader XML stream reader
+   * @return XML document string; null if no more documents in stream
+   * @throws XMLStreamException if there's an exception for the XML stream
+   */
+  public static String getSubDocument(int event, XMLStreamReader reader)
+      throws XMLStreamException {
+    int level = 0;
+    Buffer buffer = Buffer.buffer();
+    for (; reader.hasNext(); event = reader.next()) {
+      switch (event) {
+        case XMLStreamConstants.START_ELEMENT:
+          level++;
+          buffer.appendString("<").appendString(reader.getLocalName());
+          if (level == 1) {
+            String uri = reader.getNamespaceURI();
+            if (uri != null) {
+              buffer
+                  .appendString(" xmlns=\"")
+                  .appendString(uri)
+                  .appendString("\"");
+            }
+          }
+          for (int i = 0; i < reader.getAttributeCount(); i++) {
+            buffer
+                .appendString(" ")
+                .appendString(reader.getAttributeLocalName(i))
+                .appendString("=\"")
+                .appendString(encodeXmlText(reader.getAttributeValue(i)))
+                .appendString("\"");
+          }
+          buffer.appendString(">");
+          break;
+        case XMLStreamConstants.END_ELEMENT:
+          level--;
+          buffer.appendString("</").appendString(reader.getLocalName()).appendString(">");
+          if (level == 0) {
+            return buffer.toString();
+          }
+          break;
+        case XMLStreamConstants.CHARACTERS:
+          buffer.appendString(encodeXmlText(reader.getText()));
+          break;
+        default:
+      }
+    }
+    return null;
+  }
+
+  static JsonObject createIngestRecord(JsonObject marcPayload, JsonObject stylesheetResult) {
+    if (stylesheetResult.containsKey(MARC_COLLECTION)) {
+      stylesheetResult = stylesheetResult.getJsonObject(MARC_COLLECTION);
+    }
+    JsonObject inventoryPayload = stylesheetResult.getJsonObject(MARC_RECORD);
```

Comment:
This is confusing -- is MARC_RECORD a general label for "record", if so should it be renamed to RECORD_LABEL and MARC_COLLECTION to COLLECTION_LABEL?

Reply:
> This is confusing -- is MARC_RECORD a general label for "record", if so should it be renamed to RECORD_LABEL and MARC_COLLECTION to COLLECTION_LABEL?

The XML element is mapped to JSON property with same name in the stylesheets. I'll rename it.